src/test/rgw/rgw_multi/tests_ps.py

```diff
@@ -3484,13 +3500,23 @@ def test_ps_s3_creation_triggers_on_master():
 
     # start amqp receiver
     exchange = 'ex1'
-    task, receiver = create_amqp_receiver_thread(exchange, topic_name)
+    task, receiver = create_amqp_receiver_thread(exchange, topic_name, external_endpoint_address, ca_location)
     task.start()
 
     # create s3 topic
-    endpoint_address = 'amqp://' + hostname
-    endpoint_args = 'push-endpoint='+endpoint_address+'&amqp-exchange=' + exchange +'&amqp-ack-level=broker'
-    topic_conf = PSTopicS3(master_zone.conn, topic_name, zonegroup.name, endpoint_args=endpoint_args)
+    if external_endpoint_address:
+        endpoint_address = external_endpoint_address
+    elif ca_location:
+        endpoint_address = 'amqps://' + hostname
+    else:
+        endpoint_address = 'amqp://' + hostname
+    endpoint_args = 'push-endpoint='+endpoint_address+'&amqp-exchange=' + exchange +'&amqp-ack-level=broker&verify-ssl='+verify_ssl
+    if ca_location:
+        endpoint_args += '&ca-location={}'.format(ca_location)
+    if external_endpoint_address:
```

Comment:
```suggestion
    if ca_location:
```
thought that `ca_location` is the indication for secure connection?

Reply:
A secure connection to the ceph endpoint is only required when passing a rabbitmq username and password, which you will almost certainly will need for an external RabbitMQ instance.

For a RabbitMQ running on localhost with guest/guest authentication you don't need a secure connection with Ceph, even if you're going to connect to RabbitMQ over SSL.

Reply:
RGW should reject the topic creation if it sees user/password sent in cleartext

Reply:
It does, but in the test with the self-signed certificates to connect to the RabbitMQ server on localhost, I am not sending a username/password across, as the guest account will be used.